monitoring/oplog-populator/dashboard.py

```diff
@@ -0,0 +1,335 @@
+from grafanalib.core import (
+    ConstantInput,
+    DataSourceInput,
+    RowPanel,
+    Threshold,
+    Templating,
+    Template,
+)
+from grafanalib import formatunits as UNITS
+from scalgrafanalib import(
+    metrics,
+    layout,Target,
+    TimeSeries,
+    Dashboard,
+    Stat,
+    PieChart,
+)
+
+class Metrics:
+    RECONFIG_LAG = metrics.BucketMetric(
+        'oplog_populator_reconfiguration_lag_sec',
+        'job', 'connector', 'namespace'
+    ).with_defaults(
+        'job="${oplog_populator_job}"',
+        'connector=~"${connector}"',
+        'namespace="${namespace}"',
+    )
+
+    ACK_LAG = metrics.BucketMetric(
+        'oplog_populator_acknowledgement_lag_sec',
+        'opType', 'job', 'namespace'
+    ).with_defaults(
+        'job="${oplog_populator_job}"',
+        'namespace="${namespace}"',
+    )
+
+    CONNECTOR_CONFIG, BUCKETS, REQUEST_SIZE, PIPELINE_SIZE = [
+        metrics.CounterMetric(
+            name, 'job', 'namespace', *extraLabels
+        ).with_defaults(
+            'job="${oplog_populator_job}"',
+            'connector=~"${connector}"',
+            'namespace="${namespace}"',
+        ) for name, extraLabels in {
+            'oplog_populator_connector_configuration': ['connector', 'opType'],
+            'oplog_populator_connector_buckets': ['connector'],
+            'oplog_populator_connector_request_bytes': ['connector'],
+            'oplog_populator_connector_pipeline_bytes': ['connector'],
+        }.items()
+    ]
+
+    CONNECTORS = metrics.CounterMetric(
+        'oplog_populator_connectors', 'isOld',
+        'job', 'namespace'
+    ).with_defaults(
+        'job="${oplog_populator_job}"',
+        'namespace="${namespace}"',
+    )
+
+oplog_populator_up_current = Stat(
+    title='Oplog Populator Instances',
+    description='Number of active oplog populator instances',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='background',
+    noValue='0',
+    reduceCalc='last',
+    graphMode='none',
+    targets=[Target(
+        expr='\n'.join([
+            'sum(up{',
+            '    namespace="${namespace}",',
+            '    job="${oplog_populator_job}"',
+            '})',
+        ]),
+    )],
+    thresholds=[
+        Threshold('red', 0, 0.),
+        Threshold('green', 1, 1.),
+    ])
+
+kafka_connect_up_current = Stat(
+    title='Kafka Connect Instances',
+    description='Number of active kafka-connect instances',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='background',
+    noValue='0',
+    reduceCalc='last',
+    graphMode='none',
+    targets=[Target(
+        expr='\n'.join([
+            'sum(up{',
+            '    namespace="${namespace}",',
+            '    job="${kafka_connect_job}"',
+            '})',
+        ]),
+    )],
+    thresholds=[
+        Threshold('red', 0, 0.),
+        Threshold('green', 1, 1.),
```

Comment:
may be yellow if less than expected number of replicas, and green only when we reach that number?

Reply:
I don't think this is needed for this dashboard as we are not monitoring Kafka connect